Source/WebCore/rendering/RenderLayer.cpp

```diff
@@ -370,6 +370,10 @@ RenderLayer::RenderLayer(RenderLayerModelObject& renderer)
         return;
 #endif
 
+    //  We need the parent to know if we have skipped content or content-visibility root.
+    if (renderer.style().effectiveSkippedContent() && !renderer.parent())
+        return;
```

Comment:
Not really a fan of these early returns from a constructor; easy to add code at the end and make mistakes.

Reply:
You could move code inside a lambda function, and just early return from that when m_visibleContentStatusDirty gets false. That's the cleanest.

Reply:
Nice, I forgot about the lambda option, fixed.